conf/base.config

```diff
@@ -32,23 +32,23 @@ process {
   }
   $bwa{
     cpus = { check_max( 8 * task.attempt, 'cpus' ) }
-    memory = { check_max( 64.GB * task.attempt, 'memory' ) }
+    memory = { check_max( 31.GB * task.attempt, 'memory' ) }
```

Comment:
You don't need to change this in the config file - the `check_max` makes running on smaller machines easier. It looks like the documentation for this feature hasn't made it into this repository yet, but you can read it for one of the other pipelines here: https://github.com/nf-core/methylseq/blob/master/docs/usage.md#maximum-resource-requests

Quoted below for ease:

> ### Maximum resource requests
> All resource requests are checked against the following default limits (`standard` config profile shown):
> 
> ```bash
> --max_memory '128.GB'
> --max_cpus '16'
> --max_time '240.h'
> ```
> 
> If a task requests more than this amount, it will be reduced to this threshold.
> 
> To adjust these limits, specify them on the command line, eg. `--max_memory '64.GB'`.
> 
> Note that these limits are the maximum to be used _per task_. Nextflow will automatically attempt to parallelise as many jobs as possible given the available resources.

So basically, just use `--max_memory '31.GB'` for your system instead of changing anything.

Reply:
I'm sorry, this should not have be in the pull request. But thanks for the --max_memory '31.GB'

Reply:
OK, now I remember why I changed manually the code.
If I set --max_memory '31.GB' I get a Java exception. However this argument works in the rna-seq pipeline. 

```
Caused by:
  java.lang.String cannot be cast to nextflow.util.MemoryUnit
```

Reply:
Ah yeah, looks like the RNA pipeline has a [newer version](https://github.com/nf-core/rnaseq/blob/4658b260542033da339ec802311bfc88f15bcd7b/nextflow.config#L134-L165) of the `check_max` function (compared to the [chipseq version](https://github.com/nf-core/chipseq/blob/b9a495996f2292c77a9364f31bfac5d060878fec/nextflow.config#L103-L119)).

Copying and pasting the RNA function to replace the chipseq function should solve the error.

Reply:
(these kinds of consistency problems are why we're hoping to write a new tool to try to keep the core functionality of all nf-core pipelines in sync in the future!)